lib/core/test/unit/Cardano/Wallet/Primitive/AddressDerivation/MintBurnSpec.hs

```diff
@@ -0,0 +1,95 @@
+{-# LANGUAGE DataKinds #-}
+{-# LANGUAGE FlexibleInstances #-}
+
+{-# OPTIONS_GHC -Wno-orphans #-}
+
+module Cardano.Wallet.Primitive.AddressDerivation.MintBurnSpec
+    ( spec
+    ) where
+
+import Prelude
+
+import Cardano.Address.Derivation
+    ( XPrv, XPub )
+import Cardano.Address.Script
+    ( KeyHash )
+import Cardano.Wallet.Primitive.AddressDerivation
+    ( Depth (PolicyK, RootK, ScriptK)
+    , DerivationType (Hardened)
+    , Index
+    , Passphrase
+    , WalletKey (publicKey)
+    , getRawKey
+    , hashVerificationKey
+    , liftRawKey
+    )
+import Cardano.Wallet.Primitive.AddressDerivation.MintBurn
+    ( derivePolicyKey, derivePolicyPrivateKey )
+import Cardano.Wallet.Primitive.AddressDerivation.Shelley
+    ( ShelleyKey )
+import Cardano.Wallet.Primitive.AddressDerivationSpec
+    ()
+import Cardano.Wallet.Unsafe
+    ( unsafeXPrv )
+import qualified Data.ByteString as BS
+import Test.Hspec
+    ( Spec, describe, it )
+import Test.Hspec.Extra
+    ( parallel )
+import Test.QuickCheck
+    ( Arbitrary (..), Property, property, vector, (===) )
+import Test.QuickCheck.Arbitrary
+    ( arbitraryBoundedEnum )
+
+import qualified Cardano.Address.Script as CA
+
+spec :: Spec
+spec = do
+    parallel $ describe "Mint/Burn Policy key Address Derivation Properties" $ do
+        it "Policy key derivation from master key works for various indexes" $
+            property prop_keyDerivationFromXPrv
+        it "Policy public key hash matches private key" $
+            property prop_keyHashMatchesXPrv
+
```

Comment:
Perhaps one or two golden tests that you have generated manually with cardano-cli?